app/javascript/vue/components/RepositoryItemSidebar.vue

```diff
@@ -1,213 +1,280 @@
             <div @click="handleSideNavClick" id="text-item-4" class="hover:cursor-pointer text-sn-grey">QR</div>
           </div>
         </div>
-  
-        <div id="highlight-container" class="w-[1px] h-[130px] flex flex-col justify-evenly bg-sn-light-grey">
-          <div id="highlight-item-1" class="w-[5px] h-[28px] rounded-[11px]"></div>
-          <div id="highlight-item-2" class="w-[5px] h-[28px] rounded-[11px]"></div>
-          <div id="highlight-item-3" class="w-[5px] h-[28px] rounded-[11px]"></div>
-          <div id="highlight-item-4" class="w-[5px] h-[28px] rounded-[11px]"></div>
-        </div>
       </div>
-  
-      <div id="bottom" class="h-[100px] flex flex-col justify-end">
-        <div id="divider" class="w-500 bg-sn-light-grey flex px-8 items-center self-stretch h-px mb-6"></div>
-        <div id="bottom-button-wrapper" class="flex h-10 justify-end">
-          <button class="btn btn-primary">Print Label</button>
-        </div>
+
+      <div id="highlight-container" class="w-[1px] h-[130px] flex flex-col justify-evenly bg-sn-light-grey">
+        <div id="highlight-item-1" class="w-[5px] h-[28px] rounded-[11px]"></div>
+        <div id="highlight-item-2" class="w-[5px] h-[28px] rounded-[11px]"></div>
+        <div id="highlight-item-3" class="w-[5px] h-[28px] rounded-[11px]"></div>
+        <div id="highlight-item-4" class="w-[5px] h-[28px] rounded-[11px]"></div>
+      </div>
+    </div>
+
+    <div id="bottom" class="h-[100px] flex flex-col justify-end" :class="{ 'pb-6': customColumns?.length }">
+      <div id="divider" class="w-500 bg-sn-light-grey flex px-8 items-center self-stretch h-px mb-6"></div>
+      <div id="bottom-button-wrapper" class="flex h-10 justify-end">
+        <button class="btn btn-primary">Print Label</button>
       </div>
     </div>
-  
+  </div>
 </template>
-  <script>
-
-    export default {
-      name: 'RepositoryItemSidebar',
-      data() {
-        // using dummy data for now
-        return {
-            repositoryName: null,
-            defaultColumns: null,
-            customColumns: null,
-            isShowing: false,
-            sequenceExpanded: false,
-            assigned: 'Assigned to 3 private tasks that will not be displayed'
-        }
-      },
-      created() {
-        window.repositoryItemSidebarComponent = this;
-      },
-      watch: {
-        isShowing(newVal, oldVal) {
-            const element = document.getElementById('repositoryItemSidebar')
-            if (this.isShowing) {
-              element.classList.remove('translate-x-full');
-              element.classList.add('translate-x-0');
-            } else {
-              element.classList.add('transition-transform', 'ease-in-out', 'duration-300', 'transform', 'translate-x-0', 'translate-x-full');
-            }
-        }
-      },
-      mounted() {
+
+<script>
+import RepositoryStockValue from './dynamic_subcomponents/RepositoryStockValue.vue';
```

Comment:
put all components to app/javascript/vue/repository_item_sidebar/components/

Reply:
I would say something like app/javascript/vue/repository_item_sidebar/repository_values/, components is a bit too generic, since everything here is components.